cmd/notation/internal/truststore/truststore.go

```diff
@@ -0,0 +1,226 @@
+		}
+	}
+}
+
+// showCert displays details of a certificate
+func showCert(cert *x509.Certificate) {
+	fmt.Println("Issuer:", cert.Issuer)
+	fmt.Println("Subject:", cert.Subject)
+	fmt.Println("Valid from:", cert.NotBefore)
+	fmt.Println("Valid to:", cert.NotAfter)
+	fmt.Println("IsCA:", cert.IsCA)
+
+	h := sha1.Sum(cert.Raw)
+	fmt.Println("Thumbprints:", hex.EncodeToString(h[:]))
+}
+
+// DeleteAllCerts deletes all certificate files from the trust store
+// under dir truststore/x509/storeType/namedStore
+func DeleteAllCerts(storeType, namedStore string, confirmed bool, errorSlice []error) []error {
+	path, err := dir.Path.UserConfigFS.GetPath(dir.TrustStoreDir, "x509", storeType, namedStore)
+	if err == nil {
+		truststorePath := fmt.Sprintf("truststore/x509/%s/%s", storeType, namedStore)
+		prompt := fmt.Sprintf("Are you sure you want to remove all certificate files under: %q?", truststorePath)
+		confirmed, err := cmdutil.AskForConfirmation(os.Stdin, prompt, confirmed)
+		if err != nil {
+			errorSlice = append(errorSlice, fmt.Errorf("%s with error %q", path, err.Error()))
+			return errorSlice
+		}
+		if !confirmed {
+			return errorSlice
+		}
+
+		if err = os.RemoveAll(path); err != nil {
+			errorSlice = append(errorSlice, fmt.Errorf("%s with error %q", path, err.Error()))
+		} else {
+			// write out on success
+			fmt.Printf("Successfully cleared %s\n", path)
+			return nil
+		}
+	} else {
+		errorSlice = append(errorSlice, fmt.Errorf("%s with error %q", path, err.Error()))
+	}
+	return errorSlice
+}
+
+// DeleteCert deletes a specific certificate file from the
+// trust store, namely truststore/x509/storeType/namedStore/cert
+func DeleteCert(storeType, namedStore, cert string, confirmed bool, errorSlice []error) []error {
+	path, err := dir.Path.UserConfigFS.GetPath(dir.TrustStoreDir, "x509", storeType, namedStore, cert)
+	if err == nil {
+		truststorePath := fmt.Sprintf("truststore/x509/%s/%s/%s", storeType, namedStore, cert)
+		prompt := fmt.Sprintf("Are you sure you want to delete: %q?", truststorePath)
+		confirmed, err := cmdutil.AskForConfirmation(os.Stdin, prompt, confirmed)
+		if err != nil {
+			errorSlice = append(errorSlice, fmt.Errorf("%s with error %q", path, err.Error()))
+			return errorSlice
+		}
+		if !confirmed {
+			return errorSlice
+		}
+
+		if err = os.RemoveAll(path); err != nil {
+			errorSlice = append(errorSlice, fmt.Errorf("%s with error %q", path, err.Error()))
+		} else {
+			// write out on success
+			fmt.Printf("Successfully deleted %s\n", path)
+			return nil
+		}
+	} else {
+		errorSlice = append(errorSlice, fmt.Errorf("%s with error %q", path, err.Error()))
+	}
+
+	return errorSlice
+}
+
+// CheckNonErrNotExistError returns nil when no err or err is fs.ErrNotExist
+func CheckNonErrNotExistError(err error) error {
+	if err != nil && !errors.Is(err, fs.ErrNotExist) {
+		return err
+	}
+	return nil
+}
+
+// ValidateStoreType checks if storeType is supported
+func ValidateStoreType(storeType string) bool {
```

Comment:
```suggestion
func isValidStoreType(storeType string) bool {
```

Reply:
Will rename to IsValidStoreType, this func needs to be exported as it is in the truststore package. The cert package needs to use it.